mlrun/runtimes/kubejob.py

```diff
@@ -248,9 +248,29 @@ def deploy(
                 self.status.state = state
 
         if watch and not ready:
+            if not self.spec.build.secret and not self.spec.build.image.startswith("."):
+                warnings.warn(
+                    "Failed to deploy function, unable to push image to the specified docker registry."
+                    "If you wish to use the default configured registry, place '.' before the supplied image name,"
+                    "or, if you want to upload your image to dockerhub or another remote registry,"
+                    " please include a repository in the provided image."
+                    " make sure to provide a docker registry secret in your request if necessary."
```

Comment:
thinking of it again, this is completely wrong. we provide the insight with full confidence why it has failed while we could simply block the operation before it even started.

Reply:
We cannot simply block the operation because it is impossible to determine if a request is valid or invalid based on the parameters received.
If the user wishes to utilize a local registry that does not require a secret, the requested image name will not have the dot prefix and will not include a secret, but it is valid and we do not wish to prevent it.

I separated the ML-3826 and ML-3827 PRs (links below) so that the protocol fix can be merged while we decide what to do about this warning.

https://github.com/mlrun/mlrun/pull/3730